### ISSUE TITLE
Hooks inside a listener don't work

### ISSUE THREAD

**Issue by Arsik36** [FlatFilers/Guides#154]
I am following our new documentation and reached the part "Adding your first data transformation". In Replit, I navigated to fullexample.js and pasted my secret key. That code inside fullexample.js lowercases first names and validates emails. However, I don't see my first names or email validated. I made sure that I made a cURL request with the same secret that I am using in Replit. I do see listener listening to events, but I don't see hooks run. Here is my code:

```
import { webhookAction } from 'actions/webhook'
import { recordHook } from '@flatfile/plugin-record-hook'
import * as EmailValidator from 'email-validator';

/** 
 * Get your secret key at: https://platform.flatfile.com/developers and then
 * uncomment this line after setting the Secret Key value.
 */
process.env.FLATFILE_SECRET_KEY = "sk_PXhxIxbycsv99kcaGDPJj3JEaAGwP9N0"

export default function(listener) {
  // Set up your configuration here

  listener.use(
    recordHook('contacts', (record) => {
      const value = record.get('firstName').toLowerCase()
      record.set('firstName', record.get('firstName').toLowercase())
      if (!EmailValidator.validate(value)) {
        record.addError('email', 'Invalid email address')
      }
      return record
    })
  )

  listener.on('action:triggered', async (event) => {
    await webhookAction(event, 'https://webhook.site/3f74fe0a-0899-48ad-bbde-4571aaaf7ec6')
  })

}
```

**Comment by Arsik36** [comment]
Thank you @rjhyde . Your code works, but I still don't see first names getting lowercased or emails throwing errors. Do you see the same issue?

**Comment by rjhyde** [comment]
Here's a better version that does some simple checking of values and adds comments for beginners.

```listener.use(
   recordHook('contacts', (record) => {
     if (record.get('firstName')) { // check to see if firstName has a value
       const value = record.get('firstName') // put firstName into a variable
       record.set('firstName', value.toLowerCase()) // update firstName field with lowercase value
     }
     if (record.get('email')) { // check to see if email has a value
       if (!EmailValidator.validate(record.get('email'))) { // is it NOT a valid email?
         record.addError('email', 'Invalid email address')  // add an error to the field
       }
     }

     return record
   })
 )```

**Comment by elisadinsmore** [comment]
@rjhyde are either versions of the code you provided actually adding errors/correcting values in your workbook? Neither of the two options you provided does anything for me. The listener says it's updating records but my first names still contain uppercase letters and I have no errors on empty email fields.

**Comment by rjhyde** [comment]
the last code block is working for me. Try stopping and starting your replit again. The listener does some weird things with recordHook though. I see is handle the name fine but the email doesnt' get validated unless you enter that field and change it. This triggers the recordHook again but it should check it on initial pass. I'm still seeing the infinite looping too so maybe listern.use is the problem. I think we should move to a listener.on(record.*) instead.

**Comment by Arsik36** [comment]
I don't see any hooks run with the second version of your code. I tried original version and the 2 that you posted, nothing is working for me yet

**Comment by elisadinsmore** [comment]
Still not working for me, even after stopping and restarting. 

<img width="901" alt="image" src="https://user-images.githubusercontent.com/106406547/234355638-f4f7d771-6fab-4adf-86dd-d0c97e963611.png">

```
listener.use(
    recordHook('contacts', (record) => {
      if (record.get('firstName')) { // check to see if firstName has a value
        const value = record.get('firstName') // put firstName into a variable
        record.set('firstName', value.toLowerCase()) // update firstName field with lowercase value
      }
      if (record.get('email')) { // check to see if email has a value
        if (!EmailValidator.validate(record.get('email'))) { // is it NOT a valid email?
          record.addError('email', 'Invalid email address')  // add an error to the field
        }
      }

      return record
    })
  )
```

**Comment by rjhyde** [comment]
Here's what I'm getting:

https://user-images.githubusercontent.com/8014412/234355644-9b2bfae7-097e-4e18-836c-4ed8e5730d5b.mov

**Comment by elisadinsmore** [comment]
Hmm, this is what I'm getting: 

https://www.loom.com/share/69dcb4c54365484cbc648e01fb06f9ec

**Comment by Arsik36** [comment]
Same as Elisa, I am not seeing anything happen, even though I see Listener is working.

`sheet:validated` us_evt_Ke0as0PtcI6mQkWJ 4/25/2023, 5:36:36 PM
`sheet:validated` us_evt_tiDEXLu73zrIrG9m 4/25/2023, 5:36:43 PM
`records:created` us_evt_WAKzCMvQvLAHOqbm 4/25/2023, 5:36:43 PM

@ashleygmulligan2 can you advise on this? Is this a bug with the Listener, or are we not doing something right? Using docs as is doesn't show hooks to be changing values in Review stage

**Comment by elisadinsmore** [comment]
My concern here is I have no clue what to do now that it isn't working. I can see the listener is running, and something related to records:update is running, but why my code that seems valid isn't executing? I have no clue.

**Comment by rjhyde** [comment]
@elisadinsmore Ah, think I figured it out. Your video shows that your file name is full-example.js. Change that to index.js and it should work.

**Comment by elisadinsmore** [comment]
ahh that does work for the validations. May be good to call out? Though my listener is stuck in what looks like a loop after I imported a file. 

No actions are currently being taken on this sheet while I'm recording this but it's running once per second: 

https://www.loom.com/share/556968ffa3474747b86eef2bf5e10c07

**Comment by rjhyde** [comment]
Yep, I'm seeing same. I've put in another ticket for the file rename to index.js

**Comment by ashleygmulligan2** [comment]
We've fixed this now. Please try again.

**Comment by Arsik36** [comment]
@rjhyde can you paste your index.ts file here? I am still struggling to get this to work

**Comment by rjhyde** [comment]
Here's my index.js file. When I run this I don't get the results that I'm supposed to get in the docs. On manually entering a first name and taking focus off that field, it does not go to lowercase. If I reload the page it does though.
```
import { recordHook } from '@flatfile/plugin-record-hook'
import * as EmailValidator from 'email-validator';
import got from 'got';

/** 
 * Get your secret key at: https://platform.flatfile.com/developers and then
 * uncomment this line after setting the Secret Key value.
 */
process.env.FLATFILE_SECRET_KEY = "sk_LGldPcTLCNof9PZZgShGvjpZVdhOOimG"

export default function(listener) {

listener.use(
   recordHook('contacts', (record) => {
     
      const value = record.get('firstName').toString()
      if (value) {
        record.set('firstName', value.toLowerCase())
      }
      if (!EmailValidator.validate(record.get('email'))) {
        record.addError('email', 'Invalid email address')
      }
     

     return record
   })
 )

  listener.on('action:triggered', async (event) => {
    // todo: no dataUrl on this action right now
    // const json = await event.data

    // go to https://webhook.site and copy your own URL in here to test with: 
    const res = await got.post('https://webhook.site/f7cde204-9e6e-4083-a6ad-f692a08c72d4', {
      json: event.payload
    })
  })

}
```

**Comment by ashleygmulligan2** [comment]
The manual refresh after a hook runs is a bug we are fixing on the X side. Everything else should work now.

**Comment by dboskovic** [comment]
Please test this again. The UI is refreshing with changes again!

**Comment by dboskovic** [comment]
Hold on, waiting on a production deploy before it's fixed. Also this wasn't a docs bug but was an real issue with the TT reloading.

**Comment by ashleygmulligan2** [comment]
The fix has deployed to production. Thanks!
